sss_code: init at 0.2.0

krovuxdev opened the pull request:
## Description of changes
[sss](<https://github.com/SergioRibera/sss>)  is a command tools for screenshots and  code snippets.
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on pkgs/tools/misc/sss/default.nix:
```diff
@@ -0,0 +1,35 @@
+  version = "0.1.8";
+  src = fetchFromGitHub {
+    owner = "SergioRibera";
+    repo = pname;
```
```suggestion
    repo = "sss";
```

Aleksanaa commented on pkgs/tools/misc/sss/default.nix:
```diff
@@ -0,0 +1,35 @@
+  nativeBuildInputs = [pkg-config];
+  buildInputs = [fontconfig freetype libxcb];
+  meta = with lib; {
+    description = "SSS (Super ScreenShot) is a WIP: CLI/Lib to take amazing screenshot of code or screen ";
```
```suggestion
    description = "A CLI/Lib to take amazing screenshot of code or screen";
```

Aleksanaa reviewed (commented):
1. Please move to `pkgs/by-name`. Other directories needs top-level reference or package sets reference.
2. Looks like you are contributing for the first time. Please add your name to `maintainers-maintainer-list.nix` following its examples.
3. It's recommended to leave some blank lines between attributes to improve readability. You can refer to how other packages do it.

krovuxdev commented:
Hello @Aleksanaa :) 
I've addressed the three points mentioned earlier. Here's where we stand:

-  **Moved to pkgs/by-name**: Moved the package to pkgs/by-name as requested.
- **Added name to maintainers-maintainer-list.nix**: Added my name following the examples provided since it's my first 
-  **Added whitespace between attributes**: Updated the code to include whitespace between attributes for improved readability.

Please review these changes, and let me know if there's anything else I need to address.

Thanks you

Aleksanaa reviewed (commented):
Please squash all your commits leaving only two:
- maintainers: add krovuxdev
- sss: init at 0.1.8

Aleksanaa commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,42 @@
+  src = fetchFromGitHub {
+    owner = "SergioRibera";
+    repo = "sss";
+    rev = "cacc7488e303af1be3b57019ccd65d32912abb3a";
```
```suggestion
    rev = "sss_code/v${version}";
```

krovuxdev commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,42 @@
+    hash = "sha256-vB4qCLeysFYc2UwrMwj8Q/nLyygocXNN6IIP9xY1fgI=";
+  };
+
+  cargoHash = "sha256-u4unjl5Wm7Ak+JeQixweteVSk3Mc4ITXka5VhjqsRbA=";
```
Thanks for pointing that out. Should I comment out cargoHash in the code for future reference, or would you prefer to remove it altogether? Let me know what works best for the project.

Thanks again for your guidance!

Aleksanaa replied:
You should only write one, and it should be `cargoLock`.

Aleksanaa commented:
Please squash all your commits leaving only two:
- maintainers: add krovuxdev
- sss: init at 0.1.8

krovuxdev commented:
@Aleksanaa Hello, thank you for bringing it to my attention. Could you please review my commits to see if they are okay?

Aleksanaa commented:
The maintainer addition commit message is specified in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions.

krovuxdev commented:
> The maintainer addition commit message is specified in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions.

Ok, thank you, and now please, review my commits to see if they are correct

Aleksanaa commented:
@ofborg eval

Aleksanaa commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,40 @@
+  src = fetchFromGitHub {
+    owner = "SergioRibera";
+    repo = "sss";
+    rev = "sss_code/v{version}";
```
```suggestion
    rev = "sss_code/v${version}";
```

Aleksanaa commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,40 @@
+    };
+  };
+
+  nativeBuildInputs = [pkg-config];
```
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

Aleksanaa commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,40 @@
+
+  nativeBuildInputs = [pkg-config];
+
+  buildInputs = [fontconfig freetype libxcb];
```
```suggestion
  buildInputs = [ fontconfig freetype libxcb ];
```

Aleksanaa commented on pkgs/by-name/ss/sss/package.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "A CLI/Lib to take amazing screenshot of code or screen";
+    mainProgram = "sss";
+    homepage = "https://github.com/SergioRibera/sss";
+    license = with licenses; [asl20 mit];
```
```suggestion
    license = with licenses; [ asl20 mit ];
```

Aleksanaa commented:
Also your `Cargo.Lock` is different from the file in the `0.1.8` tag of the repo, but the same with one in the `main` branch. Seems to be a mistake.

krovuxdev commented:
> Also your `Cargo.Lock` is different from the file in the `0.1.8` tag of the repo, but the same with one in the `main` branch. Seems to be a mistake.

@Aleksanaa, yes, that's why I'm using "main" instead of "sss_code/v0.1.8". As you mentioned, the "sss_code/v0.1.8" is quite old. Additionally, a friend advised me to use "main" because it represents the latest version. Remember, you advised me to change the rev to "sss_code" in package.nix.

Aleksanaa commented:
> @Aleksanaa, yes, that's why I'm using "main" instead of "sss_code/v0.1.8". As you mentioned, the "sss_code/v0.1.8" is quite old. Additionally, a friend advised me to use "main" because it represents the latest version. Remember, you advised me to change the rev to "sss_code" in package.nix.

Yes, but `main` is not `0.1.8` as you wrote in the commit message. It is `0.1.8-unstable-2024-03-30`. And three months is not a long time interval. If you encounter a bug or build failure, you should use patches to backport a small part of the commit from main branch. Also using `Cargo.Lock` that is inconsistent with the author's source code will increase the probability of hidden bugs.

krovuxdev commented:
> > @Aleksanaa, yes, that's why I'm using "main" instead of "sss_code/v0.1.8". As you mentioned, the "sss_code/v0.1.8" is quite old. Additionally, a friend advised me to use "main" because it represents the latest version. Remember, you advised me to change the rev to "sss_code" in package.nix.
> 
> Yes, but `main` is not `0.1.8` as you wrote in the commit message. It is `0.1.8-unstable-2024-03-30`. And three months is not a long time interval. If you encounter a bug or build failure, you should use patches to backport a small part of the commit from main branch. Also using `Cargo.Lock` that is inconsistent with the author's source code will increase the probability of hidden bugs.


@Aleksanaa, if I set the version as "sss-unstable-2024-03-29" or "unstable-2024-03-30", is that sufficient?

Aleksanaa commented:
No. It doesn't fit the versioning guidelines in https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming. And although there is currently no policy whether to uses the `-git` version, a project that was released three months ago (and this release is a valid) is probably not within the suitable scope.

krovuxdev committed:
maintainers: add krovuxdev

krovuxdev commented:
I've updated the version of `sss_code` and also changed the title of my PR. I'll create another PR for `sss_cli` shortly. Does that sound better?

Aleksanaa commented on pkgs/by-name/ss/sss_code/package.nix:
```diff
@@ -0,0 +1,44 @@
+  };
+
+  postInstall = ''
+    rm -f $out/bin/sss_cli
```
Please add a comment here why deleting this.

krovuxdev commented:
Wait! xd

krovuxdev commented:
Done!

krovuxdev commented:
@ofborg eval

krovuxdev committed:
sss_code: init at 0.2.0

sss_code: init at 0.2.0

sss_code: init at 0.2.0

sss_code: init at 0.2.0

Aleksanaa commented:
@ofborg build

Aleksanaa commented:
@ofborg build sss_code

Aleksanaa commented:
```
error: builder for '/nix/store/i0kfrxx1cpikb5rlcl7p1262vgyvr4dr-crate-serde_repr-0.1.19.tar.gz.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://crates.io/api/v1/crates/serde_repr/0.1.19/download
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (35) Send failure: Connection reset by peer
       > error: cannot download crate-serde_repr-0.1.19.tar.gz from any mirror
       For full logs, run 'nix log /nix/store/i0kfrxx1cpikb5rlcl7p1262vgyvr4dr-crate-serde_repr-0.1.19.tar.gz.drv'.
error: builder for '/nix/store/65pz36pyb76bkvz56bnpdph1z5swdd0y-crate-tauri-winrt-notification-0.2.0.tar.gz.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://crates.io/api/v1/crates/tauri-winrt-notification/0.2.0/download
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (35) Send failure: Connection reset by peer
       > error: cannot download crate-tauri-winrt-notification-0.2.0.tar.gz from any mirror
       For full logs, run 'nix log /nix/store/65pz36pyb76bkvz56bnpdph1z5swdd0y-crate-tauri-winrt-notification-0.2.0.tar.gz.drv'.
```

@ofBorg build sss_code

krovuxdev commented:
@Aleksanaa And why this error of tauri?
sss_code does not use tauri

krovuxdev commented:
ahhh now I see, there is a tauri in cargo.lock. ok

afh commented on pkgs/by-name/ss/sss_code/package.nix:
```diff
@@ -0,0 +1,45 @@
+  buildInputs = [ fontconfig freetype libxcb ];
+
+  meta = with lib; {
+    description = "A CLI/Lib to take amazing screenshot of code";
```
According to the [recommendations for meta attributes]( https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) description should not start with an article and avoid subjective language, hence I propose to use the following shortened description from the projects readme:
```suggestion
    description = "Libraries and tools for building screenshots in a high-performance image format";
```

krovuxdev replied:
@afh it's ready. Thank you, I changed the description and also made changes to the code because the package was updated.

krovuxdev commented:
Hey @Aleksanaa  @afh   What do you think about this PR? Ready to merge, or does it need more tweaks?

krovuxdev commented on pkgs/by-name/ss/sss_code/package.nix:
```diff
@@ -0,0 +1,52 @@
+    outputHashes = {
+      "mouse_position-0.1.3" = "sha256-aYgWaozAECw+RzD2cOT7OxYzKJd2MlLsXkqW27BjSwI=";
+    };
+  };
```
@FliegendeWurst  Thank you for the suggestion, I really appreciate your help! :D

FliegendeWurst commented on pkgs/by-name/ss/sss_code/package.nix:
```diff
@@ -0,0 +1,49 @@
+
+  cargoHash = "sha256-qeDZgrGPSz+wXolZeVb2FFHjLzl1+vjzMN/3NCgaf/s=";
+
+  # Removed 'sss' command due to version differences and to separate commands.
+  postInstall = ''
+    rm -f $out/bin/sss
+  '';
```
```suggestion
  cargoBuildFlags = [ "-p" "sss_code" ];
```
Should have the same effect, and even saves some compile time.

krovuxdev replied:
I didn’t know about this, and it’s really helpful. I’ll make the change. Appreciate it!

FliegendeWurst commented on pkgs/by-name/ss/sss_code/package.nix:
```diff
@@ -0,0 +1,49 @@
+    fontconfig
+    libxcb
+  ];
+
```
```suggestion

  doCheck = false; # no tests

```
The tests take quite some time to compile otherwise.

krovuxdev replied:
@FliegendeWurst  Okay, I'll change it.
